src/utils/vcs.rs

```diff
@@ -427,8 +434,131 @@ pub fn find_heads(
     Ok(rv)
 }
 
+// Get commits from git history upto previous commit.
+// Returns a tuple of Vec<GitCommits> and the `prev_commit` if it exists in the git tree.
+pub fn get_commits_from_git<'a>(
+    repo: &'a Repository,
+    prev_commit: &str,
+) -> Result<(Vec<Commit<'a>>, Option<Commit<'a>>), Error> {
+    let mut revwalk = repo.revwalk()?;
+    revwalk.push_head()?;
+
+    let default_count = 20;
+    let commit_filter = move |id:Result<git2::Oid, git2::Error>| {
```

Comment:
I'd inline it, as it's very short filter

Reply:
@kamilogorek Markus suggested to extract the filter. Do you think inlining is better for maintainability/readability? What about the code duplication?

Reply:
Let's leave it. I only realised now that it's used in more than 1 place.